### PR TITLE
Updating docs to say TraitsUI 5 instead of 4

### PR DESCRIPTION
**Pull request by rahulporuri**
Apart from changing the 4 to 5 in a few places, changes are

- used `try`, `except` to catch IOError raised when `_version.py` is missing to provide information on how the solution
- changed the ImportError error message raised when enthought-sphinx-theme is missing to show where to download the theme from
- added a brief README.rst file.

ref: issue #260

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
```
Comment (corranwebster) [inline docs/source/conf.py]: Removing this _may_ be safe, but it was auto-generated with this coding tag, so it's probably wisest to leave it there (it signals to Python the encoding of the file).

File: docs/source/conf.py [hunk 2]
```diff
@@ -43,8 +43,15 @@
...
 d = {}
-execfile(os.path.join('..', '..', 'traitsui', '_version.py'), d)
-version = release = d['version']
+try:
+    execfile(os.path.join('..', '..', 'traitsui', '_version.py'), d)
+    version = release = d['version']
+except IOError as ioe:
+    import warnings
+    msg = '''_version.py seems to be missing,
+            which is autogenerated if the users runs
```
Comment (corranwebster) [inline docs/source/conf.py]: a clearer message would be something like "please run 'python setup.py develop' to generate this file."

File: docs/source/conf.py [hunk 3]
```diff
@@ -87,7 +94,7 @@
 
 # The name for this set of Sphinx documents.  If None, it defaults to
 # "<project> v<release> documentation".
-html_title = "TraitsUI 4 User Manual"
+html_title = "TraitsUI 5 User Manual"
```
Comment (corranwebster) [inline docs/source/conf.py]: Can we build this using the version number which is set somewhere earlier in the file?  That would prevent this particular error from ever occurring again.

Comment (rahulporuri) [inline docs/source/conf.py]: i can do some string formatting and use the version information from `execfile(...)`. would that be okay?

**Comment by corranwebster** [comment]
There may be some places it makes sense to use the `|version|` substitution in the changes that you have made to the .rst files (see http://www.sphinx-doc.org/en/stable/markup/inline.html#substitutions).

File: docs/source/conf.py (hunk 3 above)
Comment (corranwebster) [inline docs/source/conf.py]: If by the version information you mean the value you got from https://github.com/enthought/traitsui/blob/issue_260_docs/docs/source/conf.py#L48 then yes.

File: docs/source/conf.py [hunk 4]
```diff
@@ -94,7 +95,7 @@
 
 # The name for this set of Sphinx documents.  If None, it defaults to
 # "<project> v<release> documentation".
-html_title = "TraitsUI 5 User Manual"
+html_title = "TraitsUI {} User Manual".format(version.strip('.')[0])
```
Comment (corranwebster) [inline docs/source/conf.py]: `split` not `strip`, I think

File: docs/source/conf.py (hunk 3 above)
Comment (rahulporuri) [inline docs/source/conf.py]: yes. added a new commit with the aforementioned corrections/changes. will look into using |version| now

File: docs/source/conf.py (hunk 4 above)
Comment (rahulporuri) [inline docs/source/conf.py]: (ahhhhhh!!!!!! every.single.time.) sorry, corrected it!

**Comment by rahulporuri** [comment]
@corranwebster 
screen shot 2016-03-30 at 12 12 44
`|version|` seems to be behaving differently than defined [here](http://www.sphinx-doc.org/en/stable/markup/inline.html#substitutions). I guess it's because we redefine version in `source/conf.py` [here](https://github.com/enthought/traitsui/blob/issue_260_docs/docs/source/conf.py#L48). I can change this to get only `5` or maybe `5.1` if necessary.

What would be better?

**Comment by corranwebster** [comment]
It seems that the root problem is that we have both version and release set to the same value coupled with `version` being set to the `full_version` in non-released builds. See: https://github.com/enthought/traitsui/blob/master/docs/source/conf.py#L47

I would suggest that we go with a convention in the config file where we set `version` to something like "5.1" (ie. something like `version = '.'.join(d["version"].split('.', 2)[:2])`) and keep `release` as it is (ie. `release = d["version"]`).

That should work well for documentation, I think.

**Comment by corranwebster** [comment]
LGTM

**Comment by rahulporuri** [comment]
@corranwebster if you have the time, can you comment on the PR and tell me if there any more corrections/additions to make?

**Comment by corranwebster** [comment]
@rahulporuri I don't see any changes from my last comment.  Have you pushed everything?  If you have, then it's all good and can be merged.

**Comment by rahulporuri** [comment]
Ohh! I didn't want to take the liberty to merge!

**Comment by corranwebster** [comment]
It doesn't really matter who does it once the PR is approved.

**Comment by rahulporuri** [comment]
solves #272

**Pull request merged**